### ISSUE TITLE
How to turn a surface to a watertight mesh base on its minimum z value?

### ISSUE THREAD

**Issue by Grkmus** [mikedh/trimesh#768]
Hello, thanks for great library.
I have a set of points defines a mesh surface and I would like to turn it to a watertight mesh within its boundaries. The main purpose is to get the volume underneath of it, between the minimum z value and surface. Basically what I am trying to do is turning this:
![image](https://user-images.githubusercontent.com/9647128/78399684-a8580700-75f5-11ea-9ea5-3ec846799049.png)
into this:
![image](https://user-images.githubusercontent.com/9647128/78400346-cb36eb00-75f6-11ea-9c3d-d3f4af4d3d85.png)

I obtained the second picture by defining the vertices and faces manually. 

Is there any way to get something like that from the surface I have initially? 
I thought about the getting the bounding_box of the surface mesh and split it with the surface itself and get the volume under the surface. But couldn't find a method to split a watertight mesh with another mesh. (splitting by plane wouldn't work for me as I will have more complex surface mesh than this)
It is not directly issue but It would be great if someone can point me out, what to do about that.

Thanks.

**Comment by mikedh** [comment]
Oh I had some code kicking around that did something similar, I'll PR it in #769. What that PR is doing is creating a truncated triangular prism below each triangle:  
![before](https://user-images.githubusercontent.com/1403074/78455349-ce1ff300-766b-11ea-914f-8637d8559a0c.png)

Turns into:
![after](https://user-images.githubusercontent.com/1403074/78455354-d2e4a700-766b-11ea-8a4e-d298adf591b1.png)

The downside is each triangle turns into a separate watertight body, the upside is volume calculations and other metrics are correct.

**Comment by Grkmus** [comment]
Thank you very much for reply. I am so lucky that you already have that feature ready :).
Although, this is actually part of what I wanted. With this PR I am able to get volumes against a certain z value as  "above reference" and "below reference". But I think, getting volumes referenced to another mesh still not possible. 
I need something explained in this article => [article](https://www.researchgate.net/publication/263224218_A_ROBUST_ALGORITHM_TO_DETERMINE_SURFACESURFACE_INTERSECTION_IN_BOTH_PARAMETRIC_SPACES) 
I already started to work on it, to imply this approach. Will let you know if I succeed. Thanks for your help.
